common.sh

```diff
@@ -737,5 +802,34 @@ deploy_hook() {
     return 0
 }
 
+_is_rooted_in() {
+    #
+    # Check that path $1 is under "jail" $2
+    #
+    # Usage:
+    #
+    #     _is_rooted_in PATH JAIL
+    #
+    # True if PATH points inside JAIL directory. PATH does not
+    # have to exist.  Both paths are canonicalized first (i.e.
+    # symlinks are resolved paths are changed to  absolute).
+    #
+    # This is helpful to check against jailbreaking.
+    #
+    local path=$1       # PATH
+    local jail=$2       # JAIL
+    local cpath         # canonical PATH
+    local cjail         # canonical JAIL
+    local subpath       # sub-directory part of PATH
+    cpath=$(readlink -m "$path")
+    cjail=$(readlink -m "$jail")
+    subpath=${cpath#$cjail/}
+    test "$subpath" == "$cpath" && {
+        log_error "path points outside jail: $cpath not in $cjail"
```

Comment:
You're calling log_error twice, one right after the other, with very similar message. I would leave just one log_error, probably the one on line 739 - this way the caller is the one to decide what to log/print. Anyway, since the log_error message is shown to the customer in the report I'm not sure if they would understand the problem. I'm adding Jirka to review it.

Reply:
Fair enough, I'll remove the one from `_is_rooted_in()` -- it will be a "silent" function.